routes/reviews.js

```diff
@@ -0,0 +1,129 @@
+var express = require('express');
+var mongoose = require('mongoose');
+var router = express.Router();
+var reviewSchema = require('../database/review');
+var Review = mongoose.model('Review', reviewSchema);
+
+// GET Routes
+router.get('/', function(req, res, next) {
+  Review.find({}, function(err, data){
+    if(err){
+      res.status(500).send({'error': err});
+    } else {
+      if(req.get('Accept').indexOf("html") >= 0){
+        res.render('reviews', {reviews: data});
+      } else {
+        res.status(200).json(data);
+      }
+    }
+  });
+});
+router.get('/topPlaces', function(req, res, next) {
+  Review.find().sort({stars: 'desc'}).limit(3).exec(function(err, data){
+    if(err){
+      res.status(500).send({'error': err});
+    } else {
+      if(req.get('Accept').indexOf("html") >= 0){
+        res.render('reviews', {reviews: data});
+      } else {
+        res.status(200).json(data);
+      }
+    }
+  });
+});
+router.get('/new', function(req, res){
+  if(req.get('Accept').indexOf("html") >= 0){
+    res.render('newReview');
+  }
+});
+router.get('/edit/:id', function(req, res){
+  Review.find({'_id': req.params.id}, function(err, data){
+    if(err){
+      res.status(500).send({'error': err});
+    } else {
+      if(req.get('Accept').indexOf("html") >= 0) {
+        res.render('editReview', {review: data[0]});
+      }
+    }
+  });
+});
+router.get('/search', function(req, res){
+  var params = {};
+  if(req.query.name){
+    params.name = req.query.name;
+  }
+  if(req.query.placeType){
+    params.placeType = req.query.placeType;
+  }
+  if(req.query.stars){
+    params.stars = req.query.stars;
+  }
+
+  if(req.get('Accept').indexOf("html") >= 0){
+    res.render('searchReview')
+  } else {
+    Review.find(params, function(err, data){
+      if(err){
+        console.log(err);
+      } else {
+        res.status(200).json(data);
+      }
+    });
+  }
+
+});
+router.get('/:id', function(req, res, next){
+  Review.find({'_id': req.params.id}, function(err, data){
+    if(err){
+      res.status(500).send({'error': err});
+    } else {
+      if(req.get('Accept').indexOf("html") >= 0){
+    res.render('singleReview', {review: data[0]});
+  } else {
+        res.status(200).json(data);
+      }
+    }
+  });
+});
+
+// POST Routes
+router.post('/new', function(req, res, next){
+  var newReview = new Review(req.query);
+  newReview.save(function (err) {
+    if (err) return res.status(500).send({'error': err});
+  });
+  res.status(201).json({message: 'review added'});
+});
+
+// PUT Routes
+router.put('/edit/:id', function(req, res){
+  Review.update({_id: req.params.id}, req.query, function(err, data){
```

Comment:
Cette route marche très bien pour la partie HTML que t'as développé (comme pour le POST), car tu passes les modifications dans la query et pas dans le body. Cette façon de faire est un peu discutable. T'ouvres un débat entre le fait d'utiliser le body pour les données ou de s'appuyer sur le query.

Personnellement je ne suis pas pour le query car :
1. ça fait un peu SOAP et c'est une mauvais conception REST
2. on doit pouvoir définir un content-type aux données qu'on envoie.

Si tu veux lire encore sur ce débat :

http://stackoverflow.com/questions/262925/passing-params-in-the-url-when-using-http-post?lq=1
http://stackoverflow.com/questions/611906/http-post-with-url-query-parameters-good-idea-or-not

Donc depuis un client rest j'étais obligé à faire : http://localhost:3000/reviews/edit/54fb2575f9f2b5b92e0e591f?stars=3